src/routes/auth.route.ts

```diff
@@ -1,21 +1,71 @@
 import express, { Request, Response } from "express";
+import superagent from "superagent";
 
 import { register } from "../services/auth.service";
 
 const authRouter = express.Router();
 
-// authRouter.route("/twitter").get(
-// post request to Twitter API
-// receive redirectUri in response
-// redirect your user to received redirectUri res.redirect()
-// );
+const redirectUri = `http://localhost:${process.env.PORT}/api/auth/twitter-callback`;
+const twitterClientId = process.env.TWITTER_CLIENT_ID;
+const twitterClientSecret = process.env.TWITTER_CLIENT_SECRET;
+const state = {};
+const codeChallenge = "sdSD939fFFsfdfdskojilmsionjfje";
 
-// authRouter.route("/twitter_callback").get(
-// in query params: "code"
-// post reuqest to Twitter API with received "code" and "twitter_client_id", "twitter_client_secret" from .env
+// const authOptions = {
+//   method: "POST",
+//   url: "https://api.twitter.com/2/oauth2/token",
+//   headers: {
+//     "Content-Type": "application/x-www-form-urlencoded",
+//   },
+
+//   form: {
+//     grantType: "authorization_code",
+//     clientId: twitterClientId,
+//     clientSecret: twitterClientSecret,
+//     code: code,
+//     redirectUri: redirectUri,
+//   },
+// };
+
+authRouter.route("/twitter").get(async (_req: Request, res: Response) => {
+  res.redirect(
+    `https://twitter.com/i/oauth2/authorize?response_type=code&client_id=${twitterClientId}&redirect_uri=${redirectUri}&scope=tweet.read%20users.read%20follows.read%20follows.write&state=${state}&code_challenge=${codeChallenge}&code_challenge_method=plain`
+  );
+});
+
+authRouter
+  .route("/twitter-callback")
+  .get(async (req: Request, res: Response) => {
+    const code = req.query.code;
+    res.sendStatus(200);
+    if (code) {
+      try {
+        const form = {
```

Comment:
```suggestion
        const form = {
```
strange name for backend